pkg/reconciler/subscription/subscription.go

```diff
@@ -201,6 +211,20 @@ func (r *Reconciler) resolveSubscriber(ctx context.Context, subscription *v1.Sub
 		if subscriber.Ref != nil {
 			subscriber.Ref.Namespace = subscription.Namespace
 		}
+
+		// Resolve the group
+		if subscriber.Ref != nil && feature.FromContext(ctx).IsEnabled(feature.KReferenceGroup) {
+			err := subscriber.Ref.ResolveGroup(r.crdLister)
+			if err != nil {
+				logging.FromContext(ctx).Warnw("Failed to resolve Subscriber.Ref",
+					zap.Error(err),
+					zap.Any("subscriber", subscriber))
+				subscription.Status.MarkReferencesNotResolved(subscriberResolveFailed, "Failed to resolve spec.subscriber.ref: %v", err)
+				return pkgreconciler.NewEvent(corev1.EventTypeWarning, subscriberResolveFailed, "Failed to resolve spec.subscriber.ref: %w", err)
+			}
+			logging.FromContext(ctx).Debugw("Group resolved", zap.Any("spec.subscriber.ref", subscriber.Ref))
```

Comment:
I am sure we in Knative have code that prevents this method from attempting to update the spec of the resource being reconciled, but I think if any other code that does resource updates tries to use it, they are going to get unexpected spec updates because of the side effects. It would be much more clear if we did not mutate the `subscriber.Ref` field.

Reply:
I double checked this code because I wasn't sure myself, but we're not mutating the `Subscription.Subscriber.Ref`, because at the beginning of the method we deep clone the `Subscriber.Ref` destination, so we're not mutating the initial spec.